ext/io/wait/extconf.rb

```diff
@@ -6,6 +6,11 @@
 else
   target = "io/wait"
   have_func("rb_io_wait")
+
+  if RUBY_VERSION >= "3.2"
```

Comment:
Right, I was thinking it wouldn't matter because then there is already a definition for those methods, but that would be in a different process so it wouldn't mean any Ruby program already has those methods.

So the more natural thing would be to check if the method exists in the C extension code then, just like the Ruby pattern `def foo; end unless method_defined?(:foo)` but in C.